src/arm/windows/init.c

```diff
@@ -0,0 +1,252 @@
+#include <stdio.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <errno.h>
+
+#include <cpuinfo.h>
+#include <cpuinfo/internal-api.h>
+#include <cpuinfo/log.h>
+
+#include "windows_arm_init.h"
+
+/* Efficiency class = 0 means little core, while 1 means big core for now */
+#define MAX_WOA_VALID_EFFICIENCY_CLASSES		2
+#define VENDOR_NAME_MAX		CPUINFO_PACKAGE_NAME_MAX
+
+struct cpuinfo_arm_isa cpuinfo_isa;
+
+static void set_cpuinfo_isa_fields();
+static bool get_system_info_from_registry(
+	struct woa_chip_info** chip_info,
+	enum cpu_info_vendor* vendor);
+
+struct vendor_info {
+	char vendor_name[VENDOR_NAME_MAX];
+	enum cpu_info_vendor cpu_info_vendor;
+};
+
+/* Please add new vendor here! */
+static struct vendor_info vendors[] = {
+	{
+		"Qualcomm",
+		cpuinfo_vendor_qualcomm
+	}
+};
+
+/* Please add new SoC/chip info here! */
+static struct woa_chip_info woa_chips[] = {

```

Comment:
This covers literally two devices. We need a more generic detection method that doesn't require hard-coding every SoC.

Reply:
I absolutely agree that this solution is not ideal, but we haven't found any better and still robust option considering the existing Windows API. Also new SoCs won't turn up frequently, so even though it's a limited solution, it can be maintained.

Reply:
I had a similar concern but there doesn't seem to be any better way to do it with the help of Windows API.

Ampere SoC also runs Windows/Arm64 - Does that need to be added?

Reply:
Would be nice, but I can't really test that, so I'd ask to do as a follow-up commit.